### PR TITLE
Generator for rest router

### PR DESCRIPTION
**Pull request by pmeijer**
This PR also includes:
- RestRouters now have start/stop methods that are invoked at start/stop of the server.
- External rest-routers are mounted directly at their given path (`/rest/external/` is not prepended) N.B. This is the breaking change!
- Also note that the old RestComponent is removed (the format is supported but will log a warning).
- Start and stop things in correct order when starting/stopping server.

### CODE REVIEW

File: src/plugin/coreplugins/RestRouterGenerator/Templates/combine_templates.js [hunk 1]
```diff
@@ -0,0 +1,84 @@
+/*jshint node:true*/
```
Comment (kecso) [inline src/plugin/coreplugins/RestRouterGenerator/Templates/combine_templates.js]: why do we use the templates this way?
wouldn't it be more simple to require them as text?

File: src/server/middleware/ExampleRestComponent.js [hunk 2]
```diff
@@ -1,5 +1,7 @@
 /*jshint node:true*/
```
Comment (kecso) [inline src/server/middleware/ExampleRestComponent.js]: I think we can remove this file completely.

Comment (pmeijer) [inline src/server/middleware/ExampleRestComponent.js]: I'll remove it

File: src/plugin/coreplugins/RestRouterGenerator/Templates/combine_templates.js (hunk 1 above)
Comment (pmeijer) [inline src/plugin/coreplugins/RestRouterGenerator/Templates/combine_templates.js]: Yes you're right and I tried it too but ran into requirejs issues.
Looked closer at it and turns out if the name contains `.js` it throws the following error at the tests. 'Error: ENOENT: no such file or directory, open 'C:\GIT\webgme\plugin\RestRouterGenerator\RestRouterGenerator\RestRouter.js.ejs`... So I will just rename it.

**Pull request merged**